crates/bevy_ecs/src/archetype.rs

```diff
@@ -339,6 +339,9 @@ pub struct ArchetypeIdentity {
 pub struct ArchetypeComponentId(usize);
 
 impl ArchetypeComponentId {
+    // reserved index
+    pub const WORLD_METADATA: ArchetypeComponentId = ArchetypeComponentId(0);
```

Comment:
Can we just use `WORLD_METADATA` instead of `reads_all` and `writes_all`?

Reply:
No, I don't think so.

The `WorldMetadata` type is there so we can detect conflicts during system initialization and schedule building. Those are on the `ComponentId`. We definitely need that.

The `Access<ArchetypeComponentId>`  is to stop the executor from starting systems that conflict:
- `&World` needs to conflict with every other system that writes.
- `&mut World` needs to conflict with every system that reads or writes.

So we need `reads_all` and `writes_all` for those.

I actually don't remember why I added the `WORLD_METADATA` constant. Might have just put it there in case we add `&mut` versions of the metadata params.

Reply:
yeah, you're right. `&World` systems needs to conflict with `Query<&mut C>` systems, but not with `Query<C>`. We could add another bit for the "interior mutability", but that feels a little weird.